### PR TITLE
"Fix" LSP semantic token modifiers

### PR DESCRIPTION
**Pull request by davidwengier**
For purley selfish reasons, I'm trying to add support to VS for semantic token modifiers, and noticed that Roslyns legend was a bit wrong (you weren't returning the "Reassigned Variable" modifier in the legend, but _are_ returning it in the semantic token range LSP request). Additionally, Razors legend is wrong too, so exposing the fixed one so we can reuse it fixes that too.

Sneak peak!
image

### CODE REVIEW

**Comment by davidwengier** [comment]
Also, noticed that Roslyn will only ever apply one token modifier to a span. Is that intentional?

File: src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensSchema.cs [hunk 1]
```diff
@@ -125,5 +125,12 @@ public static SemanticTokensSchema LegacyTokenSchemaForRazor
 
         public static SemanticTokensSchema LegacyTokensSchemaForLSIF
             => s_vsTokenSchema;
+
+        public static string[] TokenModifiers = new string[]
+        {
+            // This must be in the same order as SemanticTokens.TokenModifiers, but skip the "None" item
+            SemanticTokenModifiers.Static,
+            nameof(SemanticTokens.TokenModifiers.ReassignedVariable)
```
Comment (dibarbet) [inline src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensSchema.cs:133]: why nameof for reassignedvariable but not the other?

**Review by dibarbet — APPROVED** [review]
> Also, noticed in the code that converts classifications to semantic tokens that Roslyn will only ever apply one token modifier to a span. Is that intentional? Or a side-effect of the fact that the current two are mutually exclusive?

I'm guessing we just never implemented it because we only returned one at the time (think this predates the existence of the reassigned variable modifier).  

The spec supports multiple, we'd just have to do the work to do the proper encoding for it.

File: src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensSchema.cs (hunk 1 above)
Comment (davidwengier) [reply src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensSchema.cs:133]: `Static` is coming from the LSP protocol DLL, as its part of the standard set. I could change it to `nameof(SemanticTokens.TokenModifiers.Static)` for consistency, but `ReassignedVariable` can't come from anywhere else, as its entirely a Roslyn construct. Between Roslyn, Razor and VS, I don't think anyone actually implements this part of the spec properly (or nobody knows how it should be implemented - myself included!)

**Comment by davidwengier** [comment]
> I'm guessing we just never implemented it 

I think actually there might be a bug in the logic for how modifiers are applied, beyond just not supporting multiple. Given that its never been used before, that is not too surprising, but will see if can fix it while I'm here.

**Pull request merged**